ci/do_ci.sh

```diff
@@ -187,9 +187,11 @@ if [ -n "$CIRCLECI" ]; then
     # As of the new coverage methodology introduced in Envoy, that has grown memory requirements too.
     # Hence we heavily reduce parallellism, to avoid being OOM killed.
     if [[ "$1" == "coverage" ]]; then
-        NUM_CPUS=4
+        NUM_CPUS=3
```

Comment:
Can we note the motivation for these changes in the PR description?

Not sure if they are related or unintentionally added to this PR.

Reply:
Sorry - this ended up here by accident.
I backed this out of here and moved part of it into https://github.com/envoyproxy/nighthawk/pull/415.
(For coverage, I want to hold off making this change: I think an upstream Envoy change may
have lowered the memory requirements).